src/turtle/env/Dmq.d

```diff
@@ -104,17 +106,20 @@ public class Dmq : Node!(DmqNode, "dmq")
 
     public void push ( T ) ( cstring channel, T data )
     {
-        // make the function work with static arrays
-        static if (is(T : cstring) || is(T : Const!(void)[]))
+        void[] serialized_data;
+
+        static if (is(T : cstring))
+            serialized_data = data.dup;
+        else static if (is(T == struct))
         {
-            auto serialized_data = cast(mstring) data.dup;
+            static if (Version.Info!(T).exists)
+                (new VersionDecorator).store(data, serialized_data);
+            else
+                Serializer.serialize(data, serialized_data);
         }
         else
-        {
-            void[] buf;
-            auto serialized_data =
-                cast(mstring) Serializer.serialize(data, buf);
-        }
+            static assert (false, "Only struct and string values are supported");
```

Comment:
Hmm, so strictly speaking this is a breaking change? (The test indicates that integers used to be supported but aren't any more.)

Reply:
Yeah it seems like the DHT code doesn't apply one-to-one here.
What about
```
static if (Version.Info!(T).exists) // implies is(T == struct)
{
    (new VersionDecorator).store(data, serialized_data);
}
else
{
    // .... the old code
}
```

Reply:
Fair point. I updated and added a test as well for versioned struct serialization and deserialization.